enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/assets/k8s/Pods.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.cloud.oracle.assets.k8s;
+
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.client.Watch;
+import io.fabric8.kubernetes.client.Watcher;
+import io.fabric8.kubernetes.client.WatcherException;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.beans.PropertyChangeSupport;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import org.netbeans.api.project.Project;
+import org.netbeans.api.project.ProjectInformation;
+import org.netbeans.api.project.ProjectUtils;
+import org.netbeans.modules.cloud.oracle.assets.OpenProjectsFinder;
+import org.openide.util.Exceptions;
+import org.openide.util.RequestProcessor;
+
+/**
+ *
+ * @author Jan Horvath
+ */
+public class Pods {
+    private final RequestProcessor RP = new RequestProcessor(Pods.class);
+    private final ClusterItem cluster;
+    private Set<String> names = null;
+    private final List<PodItem> pods = new ArrayList<> ();
+    private transient PropertyChangeSupport changeSupport = null;
+    private CountDownLatch latch = null;
+    
+    private Pods(ClusterItem cluster) {
+        this.cluster = cluster;
+    }
+    
+    public static Pods from(ClusterItem cluster) {
+        Pods instance = new Pods(cluster);
+        instance.init();
+        return instance;
+    }
+    
+    public List<PodItem> getItems() {
+        return Collections.unmodifiableList(pods);
```

Comment:
add/removePod should synchronize with this, and the client should get a copy (since this.pods can be modified further asynchronously).